vscode-lean4/src/utils/leanpkg.ts

```diff
@@ -119,11 +120,16 @@ export class LeanpkgService implements Disposable {
 
     // Return file contents with whitespace normalized.
     private readWhitespaceNormalized(fileUri: Uri) : string {
-        const contents = fs.readFileSync(fileUri.fsPath).toString();
-        // ignore whitespace changes by normalizing whitespace.
-        const re = /[ \t\r\n]+/g
-        const result = contents.replace(re, ' ');
-        return result.trim();
+        if(fsExistHelper(fileUri.fsPath)){
```

Comment:
This change *adds* an exists-read race condition, right? Regarding what to do in case the `readFile` here fails, I think we should either handle the possible exception wherever `readWhitespaceNormalized` is normalized, or change the return type to something like `string || undefined`. Returning an empty string does not distinguish between an empty file and one that doesn't exist.